web/files/paginated.go

```diff
@@ -363,3 +364,71 @@ func (f *file) IncludePath(fp vfs.FilePather) {
 	_, err := f.doc.Path(fp)
 	f.includePath = err == nil
 }
+
+// findDir is used for the result of mango requests, where only some fields can
+// have been requested
+type findDir struct {
+	*vfs.DirDoc
+	// We may want to hide some fields from the JSON response if the fields has
+	// not been requested to CouchDB, as they are blank
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+}
+
+func (d *findDir) Relationships() jsonapi.RelationshipMap { return nil }
+func (d *findDir) Included() []jsonapi.Object             { return nil }
+func (d *findDir) Links() *jsonapi.LinksList              { return nil }
+
+func newFindDir(doc *vfs.DirDoc, fields []string) *findDir {
+	dir := &findDir{doc, nil, nil}
+	if hasField(fields, "created_at") {
+		dir.CreatedAt = &doc.CreatedAt
+	}
+	if hasField(fields, "updated_at") {
+		dir.UpdatedAt = &doc.UpdatedAt
+	}
+	return dir
+}
+
+// findFile is used for the result of mango requests, where only some fields can
+// have been requested
+type findFile struct {
+	*vfs.FileDoc
+	file *file
+	// We may want to hide some fields from the JSON response if the fields has
+	// not been requested to CouchDB, as they are blank
+	CreatedAt  *time.Time `json:"created_at,omitempty"`
+	UpdatedAt  *time.Time `json:"updated_at,omitempty"`
+	Executable *bool      `json:"executable,omitempty"`
+	// Hide the internal_vfs_id and referenced_by
+	InternalID   *interface{} `json:"internal_vfs_id,omitempty"`
+	ReferencedBy *interface{} `json:"referenced_by,omitempty"`
```

Comment:
Why do we need this for files and not directories?

Reply:
For `internal_vfs_id`, this attribute only exists for files. For `referenced_by`, good question, I don't know. It is how it works today, but, yeah, looks weird. Cf https://github.com/cozy/cozy-stack/blob/b027bf9b02fe23caf0fe93bfc8aba95ba3b8ea25/web/files/paginated.go#L283